### ISSUE TITLE
[Bug]: Incorrect metadata on the access node for distributed hypertable after compression with the relation: compress_chunk_time_interval > chunk_time_interval

### ISSUE THREAD

**Issue by ivashkevichdn** [timescale/timescaledb#5815]
### What type of bug is this?

Data corruption

### What subsystems and features are affected?

Access node, Compression, Distributed hypertable

### What happened?

After compression of the distributed hypertable, the metadata on the Access Node for the FOREIGN TABLE is violated. This problem manifests itself when `compress_chunk_time_interval` has a greater value than `chunk_time_interval`.

Create a distributed hypertable
```
DROP TABLE IF EXISTS public.position_history;

CREATE TABLE public.position_history (
	create_date date NOT NULL DEFAULT CURRENT_DATE,
	keyword_id  int4 NOT NULL,
	product_id  int4 NOT NULL,
	"position"  int2 NOT NULL,
	CONSTRAINT position_history_pk PRIMARY KEY (keyword_id, product_id, create_date)
);

SELECT create_distributed_hypertable(
	'public.position_history',
	'create_date',
	partitioning_column => 'keyword_id',
	number_partitions   => 1,
	chunk_time_interval => '1day'::INTERVAL,
	replication_factor  => 1,
    data_nodes => '{"dn1"}'
);
```

Configuring distributed hypertable compression rules
```
ALTER TABLE public.position_history SET (
	timescaledb.compress,
	timescaledb.compress_orderby   = 'create_date DESC',
	timescaledb.compress_segmentby = 'keyword_id, product_id',
	timescaledb.compress_chunk_time_interval = '30day'
);
```

Let's make records in a distributed hypertable
```
INSERT INTO public.position_history(create_date, keyword_id, product_id, "position")
SELECT 
	series_create_date          AS create_date,
	series_keyword_id           AS keyword_id,
	series_product_id           AS product_id,
	round(random()*(10000-1)+1) AS "position"
FROM generate_series('2050-01-01'::date, '2050-02-15'::date, '1d'::interval) series_create_date,
	 generate_series(1000, 1010, 1) series_keyword_id,
	 generate_series(5000, 5010, 1) series_product_id;
```

Compare the number of chunks before compression in "timescaledb_information.chunks" and "chunks_detailed_size()"
```
SELECT
	count(DISTINCT i_ch.chunk_name) AS count_info_chunks,
	count(DISTINCT s_ch.chunk_name) AS count_size_chunks
FROM timescaledb_information.chunks AS i_ch, chunks_detailed_size('public.position_history') AS s_ch
WHERE i_ch.hypertable_name = 'position_history';
```
```
count_info_chunks|count_size_chunks|
-----------------+-----------------+
               46|               46|
```

Let's compress
```
SELECT compress_chunk(chunk_schema || '.' || chunk_name, if_not_compressed => True)
FROM chunks_detailed_size('public.position_history');
```
```
compress_chunk          |
------------------------+
_dist_hyper_25_397_chunk|
_dist_hyper_25_398_chunk|
_dist_hyper_25_399_chunk|
_dist_hyper_25_400_chunk|
_dist_hyper_25_401_chunk|
_dist_hyper_25_402_chunk|
_dist_hyper_25_403_chunk|
_dist_hyper_25_404_chunk|
_dist_hyper_25_405_chunk|
_dist_hyper_25_406_chunk|
_dist_hyper_25_407_chunk|
_dist_hyper_25_408_chunk|
_dist_hyper_25_409_chunk|
_dist_hyper_25_410_chunk|
_dist_hyper_25_411_chunk|
_dist_hyper_25_412_chunk|
_dist_hyper_25_413_chunk|
_dist_hyper_25_414_chunk|
_dist_hyper_25_415_chunk|
_dist_hyper_25_416_chunk|
_dist_hyper_25_417_chunk|
_dist_hyper_25_418_chunk|
_dist_hyper_25_419_chunk|
_dist_hyper_25_420_chunk|
_dist_hyper_25_421_chunk|
_dist_hyper_25_422_chunk|
_dist_hyper_25_423_chunk|
_dist_hyper_25_424_chunk|
_dist_hyper_25_425_chunk|
_dist_hyper_25_426_chunk|
_dist_hyper_25_427_chunk|
_dist_hyper_25_428_chunk|
_dist_hyper_25_429_chunk|
_dist_hyper_25_430_chunk|
_dist_hyper_25_431_chunk|
_dist_hyper_25_432_chunk|
_dist_hyper_25_433_chunk|
_dist_hyper_25_434_chunk|
_dist_hyper_25_435_chunk|
_dist_hyper_25_436_chunk|
_dist_hyper_25_437_chunk|
_dist_hyper_25_438_chunk|
_dist_hyper_25_439_chunk|
_dist_hyper_25_440_chunk|
_dist_hyper_25_441_chunk|
_dist_hyper_25_442_chunk|

```

Compare the number of chunks after compression in "timescaledb_information.chunks" and "chunks_detailed_size()"
```
SELECT
	count(DISTINCT i_ch.chunk_name) AS count_info_chunks,
	count(DISTINCT s_ch.chunk_name) AS count_size_chunks
FROM timescaledb_information.chunks AS i_ch, chunks_detailed_size('public.position_history') AS s_ch
WHERE i_ch.hypertable_name = 'position_history';
```
```
count_info_chunks|count_size_chunks|
-----------------+-----------------+
               46|                2|
```

 We see that the number of chunk records in "timescaledb_information.chunks" remains 46 as before compression, but should be 2.

 Let's run a simple query
```
SELECT *
FROM public.position_history
WHERE create_date = '2050-01-10'::date
LIMIT 10;
```
`ERROR:  [dn1]: chunk id 374 not found`

Let's look at the information about chunks
```
SELECT hypertable_schema, hypertable_name, chunk_schema, chunk_name, range_start, range_end, is_compressed
FROM timescaledb_information.chunks
WHERE hypertable_name = 'position_history';
```

```
hypertable_schema|hypertable_name |chunk_schema         |chunk_name              |range_start                  |range_end                    |is_compressed|
-----------------+----------------+---------------------+------------------------+-----------------------------+-----------------------------+-------------+
public           |position_history|_timescaledb_internal|_dist_hyper_25_397_chunk|2050-01-01 03:00:00.000 +0300|2050-01-02 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_398_chunk|2050-01-02 03:00:00.000 +0300|2050-01-03 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_399_chunk|2050-01-03 03:00:00.000 +0300|2050-01-04 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_400_chunk|2050-01-04 03:00:00.000 +0300|2050-01-05 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_401_chunk|2050-01-05 03:00:00.000 +0300|2050-01-06 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_402_chunk|2050-01-06 03:00:00.000 +0300|2050-01-07 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_403_chunk|2050-01-07 03:00:00.000 +0300|2050-01-08 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_404_chunk|2050-01-08 03:00:00.000 +0300|2050-01-09 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_405_chunk|2050-01-09 03:00:00.000 +0300|2050-01-10 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_406_chunk|2050-01-10 03:00:00.000 +0300|2050-01-11 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_407_chunk|2050-01-11 03:00:00.000 +0300|2050-01-12 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_408_chunk|2050-01-12 03:00:00.000 +0300|2050-01-13 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_409_chunk|2050-01-13 03:00:00.000 +0300|2050-01-14 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_410_chunk|2050-01-14 03:00:00.000 +0300|2050-01-15 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_411_chunk|2050-01-15 03:00:00.000 +0300|2050-01-16 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_412_chunk|2050-01-16 03:00:00.000 +0300|2050-01-17 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_413_chunk|2050-01-17 03:00:00.000 +0300|2050-01-18 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_414_chunk|2050-01-18 03:00:00.000 +0300|2050-01-19 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_415_chunk|2050-01-19 03:00:00.000 +0300|2050-01-20 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_416_chunk|2050-01-20 03:00:00.000 +0300|2050-01-21 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_417_chunk|2050-01-21 03:00:00.000 +0300|2050-01-22 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_418_chunk|2050-01-22 03:00:00.000 +0300|2050-01-23 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_419_chunk|2050-01-23 03:00:00.000 +0300|2050-01-24 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_420_chunk|2050-01-24 03:00:00.000 +0300|2050-01-25 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_421_chunk|2050-01-25 03:00:00.000 +0300|2050-01-26 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_422_chunk|2050-01-26 03:00:00.000 +0300|2050-01-27 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_423_chunk|2050-01-27 03:00:00.000 +0300|2050-01-28 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_424_chunk|2050-01-28 03:00:00.000 +0300|2050-01-29 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_425_chunk|2050-01-29 03:00:00.000 +0300|2050-01-30 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_426_chunk|2050-01-30 03:00:00.000 +0300|2050-01-31 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_427_chunk|2050-01-31 03:00:00.000 +0300|2050-02-01 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_428_chunk|2050-02-01 03:00:00.000 +0300|2050-02-02 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_429_chunk|2050-02-02 03:00:00.000 +0300|2050-02-03 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_430_chunk|2050-02-03 03:00:00.000 +0300|2050-02-04 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_431_chunk|2050-02-04 03:00:00.000 +0300|2050-02-05 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_432_chunk|2050-02-05 03:00:00.000 +0300|2050-02-06 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_433_chunk|2050-02-06 03:00:00.000 +0300|2050-02-07 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_434_chunk|2050-02-07 03:00:00.000 +0300|2050-02-08 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_435_chunk|2050-02-08 03:00:00.000 +0300|2050-02-09 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_436_chunk|2050-02-09 03:00:00.000 +0300|2050-02-10 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_437_chunk|2050-02-10 03:00:00.000 +0300|2050-02-11 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_438_chunk|2050-02-11 03:00:00.000 +0300|2050-02-12 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_439_chunk|2050-02-12 03:00:00.000 +0300|2050-02-13 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_440_chunk|2050-02-13 03:00:00.000 +0300|2050-02-14 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_441_chunk|2050-02-14 03:00:00.000 +0300|2050-02-15 03:00:00.000 +0300|true         |
public           |position_history|_timescaledb_internal|_dist_hyper_25_442_chunk|2050-02-15 03:00:00.000 +0300|2050-02-16 03:00:00.000 +0300|true         |

```

Find out the list of chunks after compression and their sizes
```
SELECT * 
FROM chunks_detailed_size('public.position_history')
ORDER BY chunk_name;
```
```
chunk_schema         |chunk_name              |table_bytes|index_bytes|toast_bytes|total_bytes|node_name|
---------------------+------------------------+-----------+-----------+-----------+-----------+---------+
_timescaledb_internal|_dist_hyper_25_397_chunk|      57344|      40960|       8192|     106496|dn1      |
_timescaledb_internal|_dist_hyper_25_427_chunk|      49152|      40960|       8192|      98304|dn1      |
```

 Let's repeat a simple request, but already defined chunk
```
SELECT *
FROM _timescaledb_internal._dist_hyper_25_397_chunk
WHERE create_date = '2050-01-10'::date
LIMIT 10;

```
```
create_date|keyword_id|product_id|position|
-----------+----------+----------+--------+
 2050-01-10|      1000|      5000|    5455|
 2050-01-10|      1000|      5001|    7470|
 2050-01-10|      1000|      5002|    7276|
 2050-01-10|      1000|      5003|    9736|
 2050-01-10|      1000|      5004|    3825|
 2050-01-10|      1000|      5005|    9177|
 2050-01-10|      1000|      5006|    4703|
 2050-01-10|      1000|      5007|    2594|
 2050-01-10|      1000|      5008|    9813|
 2050-01-10|      1000|      5009|    1680|
```

 As you can see, the request was successful.
 
 In this case, `FOREIGN TABLE "_timescaledb_internal"."_dist_hyper_25_397_chunk"` with `CONSTRAINT limited_399 CHECK (((create_date >= '2050-01-01'::date) AND (create_date < '2050-01-02'::date)) )` on the Access Node refers to the Data Node with the compressed fragment `"_timescaledb_internal". "_dist_hyper_25_397_chunk"` with `CONSTRAINT limited_364 CHECK (((create_date >= '2050-01-01'::date) AND (create_date < '2050-01-31'::date)))`

This issue manifests itself when `compress_chunk_time_interval` has a larger value than` chunk_time_interval`.
 In this example: `compress_chunk_time_interval = '30d'::interval`; `chunk_time_interval = '1d'::interval`

### TimescaleDB version affected

2.11.0

### PostgreSQL version used

14.8

### What operating system did you use?

Ubuntu 20.04 x64

### What installation method did you use?

Deb/Apt

### What platform did you run on?

On prem/Self-hosted

### How can we reproduce the bug?

```bash
-- Create a distributed hypertable
DROP TABLE IF EXISTS public.position_history;

CREATE TABLE public.position_history (
	create_date date NOT NULL DEFAULT CURRENT_DATE,
	keyword_id  int4 NOT NULL,
	product_id  int4 NOT NULL,
	"position"  int2 NOT NULL,
	CONSTRAINT position_history_pk PRIMARY KEY (keyword_id, product_id, create_date)
);

SELECT create_distributed_hypertable(
	'public.position_history',
	'create_date',
	partitioning_column => 'keyword_id',
	number_partitions   => 1,
	chunk_time_interval => '1day'::INTERVAL,
	replication_factor  => 1,
    data_nodes => '{"dn1"}'
);

-- Configuring distributed hypertable compression rules
ALTER TABLE public.position_history SET (
	timescaledb.compress,
	timescaledb.compress_orderby   = 'create_date DESC',
	timescaledb.compress_segmentby = 'keyword_id, product_id',
	timescaledb.compress_chunk_time_interval = '30day'
);

-- Let's make records in a distributed hypertable
INSERT INTO public.position_history(create_date, keyword_id, product_id, "position")
SELECT 
	series_create_date          AS create_date,
	series_keyword_id           AS keyword_id,
	series_product_id           AS product_id,
	round(random()*(10000-1)+1) AS "position"
FROM generate_series('2050-01-01'::date, '2050-02-15'::date, '1d'::interval) series_create_date,
	 generate_series(1000, 1010, 1) series_keyword_id,
	 generate_series(5000, 5010, 1) series_product_id;

-- Let's compress
SELECT compress_chunk(chunk_schema || '.' || chunk_name, if_not_compressed => True)
FROM chunks_detailed_size('public.position_history');
```

**Comment by lkshminarayanan** [comment]
Thank you @ivashkevichdn, for filing the issue!

**Comment by ivashkevichdn** [comment]
Hello!
Could you provide information, will this problem be taken for correction in the near issue? Since we have high hopes for this functionality for optimizing storage space and optimizing queries in TimescalDB.
We are really looking forward to this fix!

**Comment by jtillots** [comment]
We have had something similar happen to one of our tables. We had this:
`cr=# select * from cpu_borked limit 1;
ERROR:  [superman1]: chunk id 28433 not found
`

Also, we saw this:
`cr=# SELECT                                           
        count(DISTINCT i_ch.chunk_name) AS count_info_chunks,
        count(DISTINCT s_ch.chunk_name) AS count_size_chunks
FROM timescaledb_information.chunks AS i_ch, chunks_detailed_size('public.cpu_borked') AS s_ch
WHERE i_ch.hypertable_name = 'cpu_borked';
 count_info_chunks | count_size_chunks 
-------------------+-------------------
               108 |               107
(1 row)
`

I found the chunk that was in one list and not in the other and did a DROP FOREIGN TABLE on it. That cleared up the problem. We do have our database set as in this issue. 

Just wanted to report that we saw something similar.
